Assets/Scripts/Layers/LayerTypes/LayerNL3DBase.cs

```diff
@@ -127,6 +120,8 @@ protected virtual void OnLayerActiveInHierarchyChanged(bool activeInHierarchy)
         public LayerNL3DBase(string name)
         {
             Name = name;
+            if(this is not RootLayer) //todo: maybe move to inherited classes so this check is not needed?
+                InitializeParent();
```

Comment:
Are you trying to set the root as a default value? Because it might be more readable to pass Root as argument, or make a separate method to make the code more self-explanatory

Reply:
no, I am avoiding the Rootlayer from having its parent set, I could implement the todo, then InitializeParent() would only be called in the constructors of all derived classes. The reason I didn't do that now is that it could be easily forgotten when a new inherited classes was made, and then things would not work as expected